src/Randomizer.SMZ3.Tracking/AutoTracking/ZeldaStateChecks/FakeFlippers.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using Randomizer.Shared;
+
+namespace Randomizer.SMZ3.Tracking.AutoTracking.ZeldaStateChecks
+{
+    /// <summary>
+    /// Zelda State check for performing fake flippers
+    /// Checks if the player is in the swimming state for two states without the flippers
+    /// </summary>
+    public class FakeFlippers : ZeldaStateCheck
+    {
+        /// <summary>
+        /// Executes the check for the current state
+        /// </summary>
+        /// <param name="tracker">The tracker instance</param>
+        /// <param name="currentState">The current state in Zelda</param>
+        /// <param name="prevState">The previous state in Zelda</param>
+        public override bool ExecuteCheck(Tracker tracker, AutoTrackerZeldaState currentState, AutoTrackerZeldaState prevState)
+        {
+            if (currentState.LinkState == 0x04 && prevState.LinkState == 0x04 && tracker.Items.Any(x => x.InternalItemType == ItemType.Flippers && x.TrackingState == 0))
```

Comment:
Does this already take into account simply diving into an area to see the next screen over (e.g. Lake Hylia Island) or to do the Houlihan room, rather than fake flippers?

Reply:
Ah, right. That's a good call out since it came up in betus's stream. There's actually a separate drowning state number to theoretically should prevent this from coming up, but apparently in certain locations you can actually swim a tiny bit before drowning. Helpful for early peeking, but messes this up.

I'll need to come up with better detection for this. I'll probably tally up the number of checks in a row where swimming was detected without the flipper, and if it's over like 5 or something make the call out. That would mean that the player is swimming for a couple seconds at least.

For now I've commented out the check and return false;

Reply:
Yeah, that's another possibility. I think the only time genuine fake flippers are used is the screen transition down close to ice rod cave and the screen transition out of the guy next to the bridge? I think most others are actually more water walking (which I want to look for as well).

When I get the time, I'll look into it. Just might be a bit too busy this week for it.